This patch configures Getopt::Long to pass options
so they will be available at the second GetOptions call.

Also an option to specify the search path for libfontconfig
is added.
diff -Naur GD-2.45/Makefile.PL GD-2.45-patched/Makefile.PL
--- GD-2.45/Makefile.PL	2009-07-10 13:40:07.000000000 -0430
+++ GD-2.45-patched/Makefile.PL	2010-11-26 22:48:52.372992578 -0430
@@ -16,9 +16,9 @@
 my (@INC,@LIBPATH,@LIBS);
 my $AUTOCONFIG = 0;      # global set by try_to_autoconfigure() below
 
-my ($options,$lib_gd_path,$lib_ft_path,$lib_png_path,$lib_jpeg_path,$lib_xpm_path,$lib_zlib_path,$force);
+my ($options,$lib_gd_path,$lib_ft_path,$lib_png_path,$lib_jpeg_path,$lib_xpm_path,$lib_zlib_path,$lib_fontconfig_path,$force);
 
-use Getopt::Long;
+use Getopt::Long qw(:config pass_through);
 GetOptions("ignore_missing_gd" => \$force);
 
 unless (try_to_autoconfigure(\$options,\$lib_gd_path,\@INC,\@LIBPATH,\@LIBS) || $force) {
@@ -49,6 +49,7 @@
 			"lib_jpeg_path=s" => \$lib_jpeg_path,
 			"lib_xpm_path=s"  => \$lib_xpm_path,
 			"lib_zlib_path=s" => \$lib_zlib_path,
+                        "lib_fontconfig_path=s" => \$lib_fontconfig_path,
 		       );
 unless ($result) {
   print STDERR <<END;
@@ -64,6 +65,7 @@
      -lib_jpeg_path path            path to libjpeg
      -lib_xpm_path  path            path to libxpm
      -lib_zlib_path path            path to libpng
+     -lib_fontconfig_path path      path to libfontconfig
      -ignore_missing_gd             Ignore missing or old libgd installations and try to compile anyway
 
 If no options are passed on the command line.  The program will
@@ -111,6 +113,12 @@
     @INC     = ("-I$lib_zlib_path/include", @INC);
     @LIBPATH = ("-L$lib_zlib_path/lib", @LIBPATH); 
 }
+if( defined($lib_fontconfig_path) ) 
+{
+    print "Fontconfig library used from:     $lib_fontconfig_path\n";
+    @INC     = ("-I$lib_fontconfig_path/include", @INC);
+    @LIBPATH = ("-L$lib_fontconfig_path/lib", @LIBPATH); 
+}
 #############################################################################################
 
 if ($^O eq 'VMS'){
